src/lib.rs

```diff
@@ -33,6 +35,18 @@ pub enum Error {
 	},
 }
 
+impl fmt::Display for Error {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		match self {
+			Self::Unknown => write!(f, "Unknown error"),
+			Self::CanonicalizePath { .. } => write!(f, "Error while canonicalizing path"),
+			Self::Remove { .. } => write!(f, "Error while performing the remove operation"),
```

Comment:
I think it would be nice to print the code for these. Here's one way for handling the option:

```rust
Self::CanonicalizePath { code } => {
    let code_str = match code {
        Some(i) => format!("Error code was {}", i),
        None => "No error code was available.".into(),
    };
    write!(f, "Error while canonicalizing path. {}", code_str);
}
```